rescore/model.py

```
@@ -122,25 +131,55 @@ def matches_categorisation(self, categorisation: dso.cvss.CveCategorisation) ->
         return value == getattr(categorisation, attr)
 
 
-@dataclasses.dataclass(frozen=True)
-class CveRescoringRuleSet:
-    '''
-    Represents configuration for a single rescoring rule set.
-    A single rule set is used to perform a CVSS rescoring according to a well-defined contract.
-    One of the rule sets can be defined as default, used as fallback for rescoring if no rule set is
-    specified.
-    See rescoring documentation for more details:
+@dataclasses.dataclass
+class RuleSet[T: Rule]:
+    name: str
+    rules: list[T]
+    type: RuleSetType
+    description: str | None = None
 
-    TODO: publish rescoring-docs (removed SAP-internal reference prior to open-sourcing)
-    '''
+
+@dataclasses.dataclass
+class CveRescoringRuleSet(RuleSet[CveRescoringRule]):
+    type: RuleSetType = RuleSetType.CVE
+
+
+@dataclasses.dataclass(frozen=True)
+class DefaultRuleSetNames:
     name: str
-    description: str
-    rules: list[RescoringRule]
+    type: RuleSetType
+
+
+def find_default_rule_set_for_type(
+    default_rule_set_names: DefaultRuleSetNames,
+    rule_sets: tuple[RuleSet],
+) -> RuleSet:
+    for ruleset in rule_sets:
+        if (
+            ruleset.name == default_rule_set_names.name
+            and ruleset.type == default_rule_set_names.type
+        ):
+            return ruleset
+
+    raise ValueError('No default rule_set found for the specified type.')
```

nit: It might be helpful to also log the specified `default_rule_set_names` here. Btw, I think the name `DefaultRuleSetNames` is a little bit misleading as one instance only refers to one _name_ (!= _nameS_) and also considers the type. What about just calling it `DefaultRuleSet` or something?